compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6857,15 +6857,25 @@ private BType checkInvocationParam(BLangInvocation iExpr, AnalyzerData data) {
         int i = 0;
         BLangExpression vararg = null;
         boolean foundNamedArg = false;
+        boolean incRecordAllowAdditionalFields = incRecordParamAllowAdditionalFields != null;
         for (BLangExpression expr : iExpr.argExprs) {
             switch (expr.getKind()) {
                 case NAMED_ARGS_EXPR:
                     foundNamedArg = true;
-                    if (i < parameterCountForNamedArgs || incRecordParamAllowAdditionalFields != null) {
+                    BLangNamedArgsExpression namedArg = (BLangNamedArgsExpression) expr;
+                    boolean namedArgForIncRecordParam =
+                                  isNamedArgForIncRecordParam(namedArg.name.value, incRecordParamAllowAdditionalFields);
```

Comment:
This is the only place where `namedArg` is used right, shall we do the casting here instead of having a variable assignment.